angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -103,47 +102,85 @@ define(['angular'], function (angular) {
       }
     };
 
-    $scope.changePref = function (userPreference) {
+    $scope.cycleUnits = function (){
 
-      if (userPreference === 'C') {
-        $scope.changeToC();
-        $scope.showMetric = true;
-      } else {
-        $scope.changeToF();
+      var userPreference = $scope.nextUnits;
```

Comment:
If myPref.userWeatherPreference is K, line 91 cyleUnits is called because pref = K and curentUnits = F and nextUnits = C

So then line 106 sets userPreference to C?